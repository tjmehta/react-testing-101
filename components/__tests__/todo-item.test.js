import { shallow } from 'enzyme';

import TodoModel from '../models/todo-model';
import TodoItem from '../todo-item';

jest.unmock('../todo-item');

describe('TodoItem', () => {
    const ctx = {};

    beforeEach(() => {
        ctx.todo = new TodoModel({ id: 'id', text: 'text' });
    });

    it('should render', () => {
        const component = shallow(<TodoItem todo={ctx.todo} />);
        expect(component).toMatchSnapshot();
    });

    it('should handle done click', () => {
        const handleDoneClick = jest.fn();
        const component = shallow(
            <TodoItem onDoneClick={handleDoneClick} todo={ctx.todo} />,
        );
        component.find('button').simulate('click');
        expect(handleDoneClick).toHaveBeenCalledWith(ctx.todo);
    });
});
