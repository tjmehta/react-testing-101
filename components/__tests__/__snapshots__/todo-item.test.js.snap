// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render 1`] = `
<li>
  <span>
    text
  </span>
  <button
    onClick={[Function]}
  >
    done
  </button>
</li>
`;
